lib/surrealist/wrapper.rb

```diff
@@ -0,0 +1,101 @@
+module Surrealist
+  # A helper class for wrapphing hashes.
+  module Wrapper
+    class << self
+      # Wraps the schema hash into root/namespaces if there is a need to.
+      #
+      # @param [Object] hash to be wrapped.
+      # @param [Object] carrier instance of Carrier class that carries arguments passed to +surrealize+
+      # @param [String] klass instance's class name.
+      #
+      # @return [Hash] a wrapped hash.
+      def wrap(hash, carrier, klass = false)
+        namespaces_condition = carrier.include_namespaces || carrier.namespaces_nesting_level != DEFAULT_NESTING_LEVEL # rubocop:disable Metrics/LineLength
+
+        if !klass && (carrier.include_root || namespaces_condition)
+          Surrealist::ExceptionRaiser.raise_unknown_root!
+        end
+
+        possibly_wrapped_hash(hash, klass, carrier, namespaces_condition)
+      end
+
+      private
+
+      # Deeply copies the schema hash and wraps it if there is a need to.
+      #
+      # @param [Object] hash object to be copied.
+      # @param [String] klass instance's class name.
+      # @param [Object] carrier instance of Carrier class that carries arguments passed to +surrealize+
+      # @param [Bool] namespaces_condition whether to wrap into namespace.
+      #
+      # @return [Hash] deeply copied hash, possibly wrapped.
+      def possibly_wrapped_hash(hash, klass, carrier, namespaces_condition)
```

Comment:
Well, this also doesn't look good in fact. Could you please leave a TODO note to refactor this later?